src/agent/Core/Controller/CheckoutSession.cpp

```diff
@@ -173,12 +174,53 @@ Controller::maybeSend100Continue(Client *client, Request *req) {
 	}
 }
 
+typedef boost::tuple<Controller*, Client*, Request*> ControllerFinishInitiatingSessionCallbackPointers;
+
+void
+Controller::onWritable(EV_P_ ev_io *io, int revents) {
+        ControllerFinishInitiatingSessionCallbackPointers *cbs = static_cast<ControllerFinishInitiatingSessionCallbackPointers *>(io->data);
+	    Controller* that = cbs->get<0>();
```

Comment:
No need to use a tuple. Just pass req as data. Then you can get everything else from the req:

```c++
Request *req = static_cast<Request *>(w->data);
Client *client = static_cast<Client *>(req->client);
Controller *self = static_cast<Controller *>(Controller::getServerFromClient(client));
```